OpenRA.Mods.Cnc/Traits/TransformsNearResources.cs

```diff
@@ -0,0 +1,75 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2017 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Linq;
+using OpenRA.Mods.Common.Activities;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Cnc.Traits
+{
+	[Desc("Transforms into another actor when a resource spawns adjacent.")]
+	public class TransformsNearResourcesInfo : ITraitInfo
+	{
+		[FieldLoader.Require]
+		[ActorReference] public readonly string IntoActor = null;
+
+		public readonly CVec Offset = CVec.Zero;
+
+		[Desc("Don't render the make animation.")]
+		public readonly bool SkipMakeAnims = false;
+
+		[FieldLoader.Require]
+		[Desc("Resource type which triggers the transformation.")]
+		public readonly string Type = null;
+
+		public virtual object Create(ActorInitializer init) { return new TransformsNearResources(init.Self, this); }
+	}
+
+	public class TransformsNearResources : ITick
+	{
+		readonly TransformsNearResourcesInfo info;
+		readonly ResourceLayer resourceLayer;
+
+		public TransformsNearResources(Actor self, TransformsNearResourcesInfo info)
+		{
+			this.info = info;
+			resourceLayer = self.World.WorldActor.Trait<ResourceLayer>();
+		}
+
+		void ITick.Tick(Actor self)
+		{
+			foreach (var location in CVec.Directions.Select(d => self.Location + d))
+			{
+				var resource = resourceLayer.GetResource(location);
+				if (resource != null)
+				{
+					if (resource.Info.Type == info.Type)
```

Comment:
https://github.com/OpenRA/OpenRA/wiki/Coding-Standard suggests to "Keep functions short, and return early instead of nesting conditionals." So you probably should combine those checks to:
```
if (resource == null || resource.Info.Type != info.Type)
	continue;
```

Reply:
Changed.